src/preact/index.js

```diff
@@ -110,8 +111,9 @@ export function useLayoutEffect(effect, opt_deps) {
 }
 
 /**
- * @param {PreactDef.Context} context
- * @return {!JsonObject}
+ * @param {PreactDef.Context<T>} context
+ * @return {T|undefined}
```

Comment:
Doesn't this return the default value used when creating the context? Meaning, we should never get `undefined`, unless you used `createContext(undefined)` (which I think is an error?).

Reply:
Yeah. I think you're right. If the context allows default value, it should be reflected at the context creation time, not at the use time.